### PR TITLE
Feature/segwit

### PR DESCRIPTION
**Pull request by amougel**
Add segwit support for creating payment transaction

### CODE REVIEW

File: packages/hw-app-btc/src/Btc.js [hunk 1]
```diff
@@ -166,6 +181,32 @@ export default class Btc {
...
+    sha.update(this.serializeTransaction(transaction, true));
+    let hash = sha.digest();
+    sha = createHash("sha256");
+    sha.update(hash);
+    hash = sha.digest();
+    const look1 = indexLookup & 0xff;
+    const look2 = (indexLookup >> 8) & 0xff;
+    const look3 = (indexLookup >> 16) & 0xff;
+    const look4 = (indexLookup >> 24) & 0xff;
+    let data = Buffer.from([look1, look2, look3, look4]);
```
Comment (gre) [inline packages/hw-app-btc/src/Btc.js]: not very important but I would suggest to use `data = Buffer.alloc(4);` and then do `data.writeUInt32LE(indexLookup, 0)` . it makes it a bit less error prone and simple. 

( ultimately in the future, i wish we can find an interesting abstraction so we don't directly deal with buffer but with a descriptive specification where we express the binary format )

File: packages/hw-app-btc/src/Btc.js [hunk 2]
```diff
@@ -493,55 +544,82 @@ btc.createPaymentTransactionNew(
...
           targetTransaction.inputs[i].script =
             inputs[i].length >= 3 && typeof inputs[i][2] === "string"
               ? Buffer.from(inputs[i][2], "hex")
-              : regularOutputs[i].script;
+              : !segwit
+                ? regularOutputs[i].script
+                : Buffer.concat([
+                    Buffer.from("76a914", "hex"),
+                    this.hashPublicKey(publicKeys[i]),
+                    Buffer.from("88ac", "hex")
```
Comment (gre) [inline packages/hw-app-btc/src/Btc.js]: what are these constants? we could put them statically just so we actually have them named?

File: packages/hw-app-btc/src/Btc.js [hunk 3]
```diff
@@ -493,55 +544,82 @@ btc.createPaymentTransactionNew(
...
+                  ]);
           return this.startUntrustedHashTransactionInput(
-            firstRun,
+            !segwit ? firstRun : false,
```
Comment (gre) [inline packages/hw-app-btc/src/Btc.js]: `!segwit && firstRun`

File: packages/hw-app-btc/src/Btc.js [hunk 4]
```diff
@@ -551,12 +629,28 @@ btc.createPaymentTransactionNew(
...
-          this.serializeTransaction(targetTransaction),
-          outputScript,
-          lockTimeBuffer
+        var result = Buffer.concat([
+          this.serializeTransaction(targetTransaction, false),
+          outputScript
         ]);
 
+        if (segwit) {
+          var witness = Buffer.from("", "hex");
```
Comment (gre) [inline packages/hw-app-btc/src/Btc.js]: can just do `Buffer.alloc(0)` instead

File: packages/hw-app-btc/src/Btc.js [hunk 5]
```diff
@@ -705,13 +800,24 @@ btc.signP2SHTransaction(
...
     const inputs = [];
     const outputs = [];
+    var witness = false;
     let offset = 0;
     const transaction = Buffer.from(transactionHex, "hex");
     const version = transaction.slice(offset, offset + 4);
     offset += 4;
+    if (
+      isSegwitSupported &&
+      (transaction[offset] == 0 && transaction[offset + 1] != 0)
```
Comment (gre) [inline packages/hw-app-btc/src/Btc.js]: prefer === and !==

File: packages/hw-app-btc/src/Btc.js (hunk 2 above)
Comment (amougel) [reply packages/hw-app-btc/src/Btc.js]: These are operators for the spending script. I' ll have them named

**Pull request merged**